backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/order/sales/OrderService.java

```
@@ -0,0 +1,168 @@
+					.orgBankAccountId(targetBankAccount)
+					.tenderType(TenderType.DirectDeposit)
+					.dateAcct(dateTrx)
+					.dateTrx(dateTrx)
+					.externalId(ExternalId.ofOrNull(request.getExternalPaymentId()));
+
+			final IdentifierString orderIdentifier = IdentifierString.of(request.getOrderIdentifier());
+
+			final Optional<OrderId> orderId = resolveOrderId(orderIdentifier, orgId);
+			orderId.ifPresent(paymentBuilder::orderId);
+
+			if (!orderId.isPresent() && IdentifierString.Type.EXTERNAL_ID.equals(orderIdentifier.getType()))
+			{
+				paymentBuilder.orderExternalId(orderIdentifier.asExternalId().getValue());
+			}
+			else if (!orderId.isPresent())
+			{
+				throw new AdempiereException("No order could be found for the given order identifier!")
+						.appendParametersToMessage()
+						.setParameter("orderIdentifier", orderIdentifier.getRawIdentifierString());
+			}
+
+			paymentBuilder.createAndProcess();
+		});
+	}
+
+	private Optional<OrderId> resolveOrderId(@NonNull final IdentifierString orderIdentifier, @NonNull final OrgId orgId)
+	{
+		final OrderQuery.OrderQueryBuilder queryBuilder = OrderQuery.builder().orgId(orgId);
+
+		switch (orderIdentifier.getType())
+		{
+			case METASFRESH_ID:
+				final OrderId orderId = OrderId.ofRepoId(orderIdentifier.asMetasfreshId().getValue());
+				return Optional.of(orderId);
+			case DOC:
+				final OrderQuery getOrderByDocQuery = queryBuilder
+						.documentNo(orderIdentifier.asDoc())
+						.build();
+
+				return orderDAO.retrieveByOrderCriteria(getOrderByDocQuery)
+						.map(I_C_Order::getC_Order_ID)
+						.map(OrderId::ofRepoId);
+			case EXTERNAL_ID:
+				final OrderQuery getOrderByExternalId = queryBuilder
+						.externalId(orderIdentifier.asExternalId())
+						.build();
+
+				return orderDAO.retrieveByOrderCriteria(getOrderByExternalId)
+						.map(I_C_Order::getC_Order_ID)
+						.map(OrderId::ofRepoId);
+			default:
+				throw new AdempiereException("Given IdentifierString type is not supported!")
```

please throw an InvalidIdentifierException (subclass of AdempiereException)